chromium/rules.js

```diff
@@ -405,51 +416,61 @@ RuleSets.prototype = {
    */
   potentiallyApplicableRulesets: function(host) {
     // Have we cached this result? If so, return it!
-    var cached_item = this.ruleCache.get(host);
-    if (cached_item !== undefined) {
-      log(DBUG, "Ruleset cache hit for " + host + " items:" + cached_item.length);
+    if (this.ruleCache.has(host)) {
+      let cached_item = this.ruleCache.get(host);
+      let len = cached_item && cached_item != nullIterable ? cached_item.size : 0;
+      log(DBUG, "Ruleset cache hit for " + host + " items:" + len);
       return cached_item;
+    } else {
+      log(DBUG, "Ruleset cache miss for " + host);
     }
-    log(DBUG, "Ruleset cache miss for " + host);
 
-    var results = [];
+    // Let's search!
+    let resultSet = new Set();
+
     if (this.targets.has(host)) {
       // Copy the host targets so we don't modify them.
-      results = results.concat(this.targets.get(host));
+      let results = this.targets.get(host);
+      results.map(result => resultSet.add(result));
     }
 
     // Ensure host is well-formed (RFC 1035)
-    if (host.indexOf("..") != -1 || host.length > 255) {
-      log(WARN,"Malformed host passed to potentiallyApplicableRulesets: " + host);
-      return null;
+    if (host.length > 255 || host.indexOf("..") != -1) {
+      log(WARN, "Malformed host passed to potentiallyApplicableRulesets: " + host);
+      return nullIterable;
     }
 
     // Replace each portion of the domain with a * in turn
-    var segmented = host.split(".");
-    for (let i=0; i < segmented.length; i++) {
-      let tmp = segmented[i];
-      segmented[i] = "*";
-      results = results.concat(this.targets.get(segmented.join(".")));
-      segmented[i] = tmp;
+    let parts = host.split(".");
+    for (let i = 0; i < parts.length; i++) {
+      let tmp = parts[i];
+      parts[i] = "*";
+
+      let domain = parts.join(".");
+      let results = this.targets.get(domain);
+      if (results) {
+        results.map(result => resultSet.add(result));
+      }
+
+      parts[i] = tmp;
     }
+
     // now eat away from the left, with *, so that for x.y.z.google.com we
     // check *.z.google.com and *.google.com (we did *.y.z.google.com above)
-    for (var i = 2; i <= segmented.length - 2; ++i) {
-      var t = "*." + segmented.slice(i,segmented.length).join(".");
-      results = results.concat(this.targets.get(t));
```

Comment:
Please revert this rename. Then just use one of the methods I suggested above to update the `results`